modules/ArchiveLibrary/Folder/Presenters/FolderPresenter.php

```
@@ -101,4 +144,24 @@ private function getLastAudit(): ?array
 
         return $lastAudit ? (new AuditPresenter($lastAudit))->getData() : null;
     }
+
+    private function getFolderSize(): int
+    {
+        // Use cache if available
+        if (self::$fileSizesCache !== null) {
+            return self::$fileSizesCache[$this->folder->id] ?? 0;
+        }
+
+        // Fallback to direct query if cache not primed (single item presentation)
+        $totalSize = \Modules\ArchiveLibrary\File\Models\File::where('folder_id', $this->folder->id)
+            ->with(['media' => function ($query) {
+                $query->select('id', 'model_id', 'size');
+            }])
+            ->get()
+            ->sum(function ($file) {
+                return $file->media->sum('size');
+            });
+
+        return (int) $totalSize;
```

The type casting `(int)` might silently truncate large folder sizes. Consider adding validation or logging if `$totalSize` exceeds PHP_INT_MAX.